### DISCUSSION TITLE
Migration from Keycloak to Azure

### DISCUSSION THREAD

**Discussion by stereo9** [kimai/kimai#4739] [Questions & Answers]
Hi,
just a quick question to avoid problems.
I am planning switch authentication methods from keycloak to entra / azure.

Is there anything that should be considered in terms how Kimai handles authentication?
or will it work seamlessly because everything is mapped to the email adress anyways?

**Comment by kevinpapst** [comment]
If you use the same ID (email) in Keycloak and Azure and Kimai, then you are good to go. 
Kimai just checks the ID field and maps the user.

**Reply by stereo9** [reply]
wow, thanks for the blazing fast reply.
will report how it worked out!
